app/src/main/java/app/grapheneos/camera/ui/composable/components/TopBarActions.kt

```diff
@@ -0,0 +1,172 @@
+
+// 40.0.dp was taken from an constant internal to the material library
+// IconButtonTokens.StateLayerSize used by IconButton's code internally
+private val SPACE_PER_ICON_BUTTON = 40.dp
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+fun TopBarActions(
+    actions : Iterable<TopBarAction>,
+    onActionClicked: (id: Any) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+
+    var size by remember {
+        mutableStateOf(IntSize.Zero)
+    }
+
+    var dropDownMenuExpanded by remember {
+        mutableStateOf(false)
+    }
+
+    Box (
+        modifier = Modifier
+            .then(modifier)
+            .onGloballyPositioned {
+                size = it.size
+            },
+        contentAlignment = Alignment.CenterEnd
+    ) {
+        if (size == IntSize.Zero) return@Box
+
+        val width = size.width
+
+        // Subtract 1 for the more options icon
+        val maxVisibleActions = (width / SPACE_PER_ICON_BUTTON.toPx()).toInt() - 1
+
+        // Ensure there is space to at least have the more options icon
+        assert(maxVisibleActions >= 0)
+        { "Please ensure that TopBarActions gets a width of 40.dp at least" }
+
+        val visibleActions = arrayListOf<TopBarAction>()
+        val moreOptionsActions = arrayListOf<TopBarAction>()
+
+        for (action in actions) {
+            if (action.alwaysInMoreOptions || visibleActions.size >= maxVisibleActions) {
+                moreOptionsActions.add(action)
+            } else {
+                visibleActions.add(action)
+            }
+        }
+
+        Row (
+            horizontalArrangement = Arrangement.End,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            for (visibleAction in visibleActions) {
+                QuickTooltip(
+                    message = visibleAction.title
+                ) {
+                    IconButton(onClick = {
+                        onActionClicked(visibleAction.id)
+                    }) {
+                        Icon(visibleAction.icon, visibleAction.title)
+                    }
+                }
+
+            }
+
+            Box {
+                if (moreOptionsActions.isNotEmpty()) {
+                    QuickTooltip(
+                        message = "More Options"
+                    ) {
+                        IconButton(onClick = {
+                            dropDownMenuExpanded = !dropDownMenuExpanded
+                        }) {
+                            Icon(
+                                Icons.Filled.MoreVert,
+                                "More Options"
+                            )
+                        }
+                    }
+
+                    DropdownMenu(
+                        expanded = dropDownMenuExpanded,
+                        onDismissRequest = {
+                            dropDownMenuExpanded = false
+                        },
+                        modifier = Modifier.width(200.dp)
```

Comment:
It's not guaranteed that screen width is over 200 dp, it can be less than that on smaller devices when screen scaling is used.

Reply:
Ah that makes sense, will update the thread once I push the changes for the same

Reply:
We're filling up a certain ratio of the screen width based on the screen in which the top bar is situated